cli/azd/pkg/auth/manager.go

```diff
@@ -1152,8 +1156,37 @@ type LogInDetails struct {
 	Account   string
 }
 
-// LogInDetails method for Manager to return login details
+// LogInDetails contains information about the currently logged in user.
+// It provides details about the type of login (email-based or client ID-based)
+// and the account identifier. When legacy authentication is used, it will
+// return the account name from the az CLI.
 func (m *Manager) LogInDetails(ctx context.Context) (*LogInDetails, error) {
+	userConfig, err := m.userConfigManager.Load()
+	if err != nil {
+		return nil, fmt.Errorf("reading user config: %w", err)
+	}
+
+	if shouldUseLegacyAuth(userConfig) {
+		log.Printf("delegating auth to az since %s is set to true", useAzCliAuthKey)
+
+		if err := m.azCli.CheckInstalled(); err != nil {
+			return nil, fmt.Errorf("checking az cli installation (using legacy auth): %w", err)
+		}
+
+		logInType := EmailLoginType
+		azAccount, err := m.azCli.Account(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("fetching az cli account: %w", err)
+		}
+		if azAccount.User.Type != "user" {
```

Comment:
This looks working to me based on az's implementation of [_profile.py](https://github.com/Azure/azure-cli/blob/e9f92bc98de1c123317ae3cbdbbc065b16185502/src/azure-cli-core/azure/cli/core/_profile.py#L595)